dipy/reconst/peak_direction_getter.pyx

```diff
@@ -75,8 +75,9 @@ cdef class PeaksAndMetricsDirectionGetter(DirectionGetter):
                 raise IndexError("point outside data")
 
         # Check to see how many peaks were found in the voxel
-        for numpeaks in range(self._ind.shape[3]):
-            if self._ind[ijk[0], ijk[1], ijk[2], numpeaks] < 0:
+        for numpeaks in range(self._ind.shape[3] + 1):
+            if (numpeaks >= self._ind.shape[3] or
```

Comment:
I am not sure that I follow this. You add one more item into the iteration (the highest value of the  shape on  the last dimension), but then you seem to ignore this value? Is that really supposed to be a `>=`, not just `>`?

Reply:
The problem was when numpeaks==self._ind.shape[3]. 
E.g. if the true number of peaks is 3 and self._ind.shape[3]==3, the loop was exiting with numpeaks==2 (never entering the if), later on, this was used in range(numpeaks), which then miss the last existing peak.

My fix didn't make the code clear. I re-worked it following your comment, making the logic more clear.

Thanks @arokem